lib/Dialect/SPIRV/SPIRVOps.cpp

```diff
@@ -199,6 +199,162 @@ static void printNoIOOp(Operation *op, OpAsmPrinter *printer) {
+        if (!resultPair.second) {
+          return parser->emitError(indicesLocation, resultPair.first);
+        }
+        if (index < 0 ||
+            static_cast<uint64_t>(index) >= cType.getNumElements()) {
+          return parser->emitError(indicesLocation, "index ")
+                 << index << " out of bounds for " << type;
+        }
+      }
+      resultType = cType.getElementType(index);
+    } else {
+      return parser->emitError(indicesLocation,
+                               "cannot extract from non-composite type ")
+             << resultType << " with index " << index;
+    }
+  }
+
+  // Result type must be a pointer with the same StorageClass.
+  state->addTypes(spirv::PointerType::get(resultType, resultStorageClass));
+  return success();
+}
+
+static void print(spirv::AccessChainOp op, OpAsmPrinter *printer) {
+  *printer << spirv::AccessChainOp::getOperationName() << ' ' << *op.base_ptr()
+           << '[';
+  printer->printOperands(op.indices());
+  *printer << "] : " << op.base_ptr()->getType();
+}
+
+static LogicalResult verify(spirv::AccessChainOp accessChainOp) {
+  auto resultType = accessChainOp.base_ptr()
+                        ->getType()
+                        .dyn_cast<spirv::PointerType>()
+                        .getPointeeType();
+  uint64_t indexCount = 0;
+  int32_t index = 0;
+
+  for (auto indexSSA : accessChainOp.indices()) {
+    if (auto cType = resultType.dyn_cast<spirv::CompositeType>()) {
+      index = 0;
+      if (resultType.isa<spirv::StructType>()) {
+        Operation *op = indexSSA->getDefiningOp();
+        if (!op) {
+          return accessChainOp.emitOpError(
+              "index must be an integer constant to access "
+              "element of StructType");
+        }
+        auto resultPair = extractValueFromConstOp(op, index);
+        if (!resultPair.second) {
+          return accessChainOp.emitOpError(resultPair.first);
+        }
+        if (index < 0 || static_cast<uint64_t>(index) >= cType.getNumElements())
+          return accessChainOp.emitOpError("index ")
+                 << index << " out of bounds for " << resultType;
+      }
+      resultType = cType.getElementType(index);
+    } else {
+      return accessChainOp.emitOpError(
+                 "cannot extract from non-composite type ")
+             << resultType << " with index " << index;
+    }
+    ++indexCount;
+  }
+
+  if (!indexCount) {
+    return accessChainOp.emitOpError(
+        "expected at least one index for spv.AccessChain");
+  }
+
+  auto providedResultType =
+      accessChainOp.getType().dyn_cast<spirv::PointerType>();
+  if (!providedResultType) {
+    return accessChainOp.emitOpError(
+               "result type must be a pointer, but provided")
+           << providedResultType;
+  }
+
+  if (resultType != providedResultType.getPointeeType()) {
+    return accessChainOp.emitOpError(
+               "invalid result type: expected a pointer to ")
```

Comment:
The type itself should print as a pointer type; so I think we don't need the `a pointer to` part in the message.

Reply:
Updated.